content/en/blog/2022-05-10-may-2022-security-announcement.md

```diff
@@ -0,0 +1,288 @@
+
+If you cannot immediately update or are hard pressed for time and need a
+work-around for now, please see the CVE advisories linked above for more
+information.
+
+## Some Background
+
+Last week the Flux Security Team disclosed three new vulnerabilities
+which affect most versions below v0.29, and have a greater impact in
+multi-tenancy deployments.
+
+The reason why the impact is greater in multi-tenancy deployments is due
+to the way that Flux/GitOps works. Flux tends to operate like a cluster
+admin, having permissions to apply any changes to a cluster, regardless
+of their scope - at namespace or cluster levels. Users having access to
+a source repository, or simply having access to create/alter Flux
+objects within a cluster can instruct Flux to apply such changes, which
+in single tenancy effectively means that such users have cluster admin
+permissions to the target clusters. The caveat being that using Flux you
+can add additional security controls between the user and the target
+cluster, for example, before each change is merged into a repository
+Pull Requests must be created requiring peer reviews. The Open GitOps
+community started defining and codifying this further - have a look [at
+this blog post](https://opengitops.dev/blog/sec-gitops/) if
+you want to know more.
+
+In multi-tenancy environments, users with similar permissions can only
+affect part of a cluster, or an isolated cluster in a group of clusters.
+Therefore, if a user can privilege escalate (or deny service) at a Flux
+level, it will have an impact way larger than on single tenancies, as
+the users in a way already had such permissions - maybe with some
+balances and checks.
+
+At time of writing, we are unaware of any public exploits in the wild,
+and therefore have no reason to believe such vulnerabilities have been
+actively exploited.
+
+## The advisories in detail
+
+### [CVE-2022-24817](https://github.com/fluxcd/flux2/security/advisories/GHSA-vvmq-fwmg-2gjc) - Kubeconfig Validation
+
+At the beginning of the Flux2 journey we implemented a feature to apply
+state to remote clusters. It enables users to have a management cluster
+in which Flux is installed, which then applies changes to other
+clusters, making it ideal for some multi-tenancy scenarios in which high
+isolation across tenants is needed.
+
+The connection between the management cluster and the target clusters is
+done via a kubeconfig which is set at the `spec.KubeConfig` field in
+either `Kustomization` or `HelmRelease` objects.
+
+One interesting thing about kubeconfigs is that they are quite
+extensible. You can for example define an executable which will then be
+automatically called by `kubectl` every time it requires a new on-demand
+access token. An example of this in action is
+[aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator),
+which enables AWS users to authenticate against AWS and reuse JWT tokens
+to access EKS clusters seamlessly.
```

Comment:
Not sure the initial attempt was clear enough to folks that are not aware of its ins and outs:
```suggestion
which enables AWS users to authenticate against AWS and then use the
returning JWT tokens to access their EKS clusters. Once the token expires,
the process happens again. All that is managed by `kubectl` behind the scenes
without user intervention.
```